pkg/monitor/cluster/podconditions.go

```diff
@@ -160,3 +161,36 @@ func (mon *Monitor) _emitPodContainerRestartCounter(ps *corev1.PodList) {
 		}
 	}
 }
+
+func (mon *Monitor) _emitOOMKilledPods(ps *corev1.PodList) {
+	for _, p := range ps.Items {
+		if !namespace.IsOpenShiftNamespace(p.Namespace) {
+			continue
+		}
+
+		// pod status doesn't matter because pod could be running but the container might be oomkilled
+		for _, cntrStatus := range p.Status.ContainerStatuses {
+			if cntrStatus.State.Terminated == nil {
+				continue
+			}
+			if cntrStatus.State.Terminated.Reason == "OOMKilled" {
+				mon.emitGauge("pod.oomkilled", 1, map[string]string{
+					"name":          p.Name,
+					"namespace":     p.Namespace,
+					"nodeName":      p.Spec.NodeName,
+					"containername": cntrStatus.Name,
```

Comment:
Do you think it make sense if we add the pod name as a dimension?

Reply:
@wanghaoran1988  `p.Name` represents pod name which is already added to the metric.